### PR TITLE
Add WordPress primitives package

### PR DESCRIPTION
**Pull request by youknowriad**
Extracted from #17055 

These primitives are useful in packages that don't depend on the components package. To avoid circular dependencies, these are best extracted to their own package. Potentially they could make sense in the "element" package as well.

### CODE REVIEW

File: docs/rfc/block-registration.md [hunk 1]
```diff
@@ -173,7 +173,7 @@ Setting `parent` lets a block require that it is only available when nested with
 
 An icon property should be specified to make it easier to identify a block. These can be any of WordPress' Dashicons (slug serving also as a fallback in non-js contexts).
 
-**Note:** It's also possible to override this property on the client-side with the source of the SVG element. In addition, this property can be defined with JavaScript as an object containing background and foreground colors. This colors will appear with the icon when they are applicable e.g.: in the inserter. Custom SVG icons are automatically wrapped in the [wp.components.SVG](/packages/components/src/primitives/svg/README.md) component to add accessibility attributes (aria-hidden, role, and focusable).
+**Note:** It's also possible to override this property on the client-side with the source of the SVG element. In addition, this property can be defined with JavaScript as an object containing background and foreground colors. This colors will appear with the icon when they are applicable e.g.: in the inserter. Custom SVG icons are automatically wrapped in the [wp.components.SVG](/packages/primitives/src/svg/README.md) component to add accessibility attributes (aria-hidden, role, and focusable).
```
Comment (gziolo) [inline docs/rfc/block-registration.md]: Should it be `wp.primitives.SVG` now?

File: packages/primitives/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+	"name": "@wordpress/primitives",
+	"version": "0.0.1",
+	"description": "WordPress cross-plateform primitives.",
```
Comment (gziolo) [inline packages/primitives/package.json]:
```suggestion
	"description": "WordPress cross-platform primitives.",
```

File: packages/primitives/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	"author": "The WordPress Contributors",
+	"license": "GPL-2.0-or-later",
+	"keywords": [
+		"wordpress",
+		"browser",
+		"async"
```
Comment (gziolo) [inline packages/primitives/package.json]: Keywords need some polishing 😅

Comment (youknowriad) [reply packages/primitives/package.json]: lol

**Review by gziolo — APPROVED** [review]
It looks good for me but I would appreciate some second opinion. I always wanted to extract such package so I'm biased :)

I haven't tested this PR but it all looks straightforward and Travis will catch issues. All components are re-exported in `@wordpress/components for backward compatibility. We should revisit it later to decide whether to deprecate them in components package.

**Comment by pinarol** [comment]
Looks good! We have been doing some experiments with @dratwas about x-platform primitives and I was already wondering what could be a better place to keep them since they don't quite belong to `@wordpress/components'.

File: package-lock.json [hunk 4]
```diff
@@ -24686,7 +24695,7 @@
 			"dependencies": {
 				"clone-deep": {
 					"version": "0.2.4",
-					"resolved": "https://registry.npmjs.org/clone-deep/-/clone-deep-0.2.4.tgz",
+					"resolved": "http://registry.npmjs.org/clone-deep/-/clone-deep-0.2.4.tgz",
```
Comment (aduth) [inline package-lock.json]: Back and forth we go 😵 (I see local changes on `master` when I run `npm install` using the latest version of NPM)

Comment (youknowriad) [reply package-lock.json]: 😓

**Pull request merged**